### ISSUE TITLE
helm chart error: expected '::' in provider reference

### ISSUE THREAD

**Issue by timmyers** [pulumi/pulumi-kubernetes#1149]
### Problem description

Getting error trying to instantiate helm chart

- I saw an issue in the following Pulumi Program:

```typescript
const cluster = new ocean.KubernetesCluster(name, {
      name,
      region: 'sfo2',
      version: '1.17.5-do.0',
      nodePool: {
        name: 'cloudwaste1',
        size: 's-1vcpu-2gb',
        autoScale: true,
        minNodes: 1,
        maxNodes: 3,
      },
    }, defaultOpts);

    const k8sProvider = new k8s.Provider('k8s', {
      kubeconfig: cluster.kubeConfigs[0].rawConfig,
    }, defaultOpts);

    const k8sDefaultOpts: pulumi.ComponentResourceOptions = {
      ...defaultOpts,
      provider: k8sProvider,
    };

    const _ = new k8s.helm.v3.Chart('metrics-server', {
      repo: 'stable',
      chart: 'metrics-server',
      version: '2.11.1',
      namespace: 'kube-system',
      values: {
        args: [
          '--logtostderr',
          '--kubelet-preferred-address-types=InternalIP',
          '--cert-dir=/tmp',
          '--kubelet-insecure-tls',
          '--secure-port=8443',
        ],
      },
    }, k8sDefaultOpts);
```

### Errors & Logs

```
Diagnostics:
  kubernetes:helm.sh:Chart (metrics-server):
    error: expected '::' in provider reference ''
```

### Affected product version(s)

pulumi CLI `v2.3.0`
@pulumi/kubernetes `v2.2.2` (seems to happen with older versions too)

**Comment by leezen** [comment]
I can't reproduce this locally (so not using a DO cluster, but am providing kubeconfig). Do you mind sharing what's in your `defaultOpts`?

**Comment by timmyers** [comment]
I just figured this out.  I had a stack that was originally implemented with golang (including this helm chart).

I reached the unfortunate conclusion that golang was just too painful to use for pulumi and decided to switch back to typescript.

Reimplementing the code to have no diffs somehow resulted in this error, but if I had pulumi destroy the helm chart, and then recreate it with the typescript program it worked fine.   Unfortunately I don't have the state file from when the go program was controlling the stack.

Feel free to close, I think this is a pretty extreme edge case.
